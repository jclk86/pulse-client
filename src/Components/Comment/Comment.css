.container_comments {
  margin: 0 auto;
  max-width: 800px;
}

.comment_username_link {
  text-decoration: none;
  color: rgb(51, 160, 255);
  font-weight: bold;
}

.container_comment_content {
  border: 1px solid gray;
  margin-top: 10px;
  border-radius: 10px;
}

.comment_content {
  font-size: 20px;
}

.lights_off_comments_content {
  background-color: rgb(59, 59, 59);
}

.comment_author_info {
  width: 100%;
  margin-right: auto;
  text-align: left;
  padding-left: 5px;
  border-bottom: 1px solid gray;
}

.list_comment_item {
  list-style: none;
}

.hide {
  display: none;
}

.list_comment_item:hover .hide {
  display: block;
}

.edit_comment_btn,
.delete_comment_btn {
  margin: 5px;
  min-width: 150px;
  background-color: rgb(51, 160, 255);
  border-radius: 3px;
  border: none;
  color: white;
  font-weight: bold;
  font-size: 16px;
}

.edit_comment_btn:hover,
.delete_comment_btn:hover {
  cursor: pointer;
}
