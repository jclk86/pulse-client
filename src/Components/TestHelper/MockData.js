export const mockUser = {
  username: "Pop123",
  password: "Password123!",
  fullname: "Bob Smith",
  email: "BSmith@gmail.com",
  image_url:
    "https://images.pexels.com/photos/1622419/pexels-photo-1622419.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
  location: "TEST, LOCATION",
  profile: "test profile"
};

export const mockArticle = {
  title: "test article",
  content: "test content",
  article_category: "News",
  id: 1,
  date_created: "2019-10-14T21:49:20.570Z",
  author: {
    username: mockUser.username
  }
};

export const mockComment = {
  id: 1,
  content: "test comment content",
  date_created: "2019-10-14T21:49:20.570Z",
  article_id: 1,
  user: {
    id: 1,
    username: mockArticle.author.username
  }
};
