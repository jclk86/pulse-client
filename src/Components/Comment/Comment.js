import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import TokenService from "../../Services/token-service";
import EditCommentForm from "../EditCommentForm/EditCommentForm";
import CommentService from "../../Services/comment-api-service";
import ArticleContext from "../../Context/ArticleContext";
import { DateFormatter } from "../Utils/Utils";
import PropTypes from "prop-types";
import "./Comment.css";

class Comment extends Component {
  static contextType = ArticleContext;
  constructor(props) {
    super(props);
    this.state = {
      isEditing: false
    };
  }

  handleEditClick = () => {
    this.setState({ isEditing: !this.state.isEditing });
  };

  handleDeleteClick = comment_id => {
    CommentService.deleteComment(comment_id).then(() =>
      this.context.deleteComment(comment_id)
    );
  };

  render() {
    const token = TokenService.readJwtToken();
    const { comment } = this.props;

    return (
      <li className="list_comment_item" key={comment.id}>
        {this.state.isEditing ? (
          <EditCommentForm
            comment={comment}
            handleEditClick={this.handleEditClick}
          ></EditCommentForm>
        ) : (
          <div className="container_comment_content">
            <p className="comment_author_info">
              <span>{comment.user.username} on </span>{" "}
              <span>{DateFormatter(comment.date_created)}</span>
            </p>
            <p className="comment_content">{comment.content}</p>
            {token.user_id === comment.user.id ? (
              <div className="container_comment_btn hide">
                <button
                  type="button"
                  className="edit_comment_btn"
                  onClick={this.handleEditClick}
                >
                  Edit
                </button>
                <button
                  type="button"
                  className="delete_comment_btn"
                  onClick={() => this.handleDeleteClick(comment.id)}
                >
                  Delete
                </button>
              </div>
            ) : null}
          </div>
        )}
      </li>
    );
  }
}

Comment.propTypes = {
  comment: PropTypes.shape({
    content: PropTypes.string,
    user: PropTypes.shape({
      id: PropTypes.number
    })
  })
};

export default withRouter(Comment);
