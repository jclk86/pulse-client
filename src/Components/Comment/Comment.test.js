import React from "react";
import { BrowserRouter } from "react-router-dom";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AuthApiService from "../../Services/auth-api-service";
import ArticleApiService from "../../Services/article-api-service";
import CommentApiService from "../../Services/comment-api-service";
import { mockUser, mockArticle, mockComment } from "../TestHelper/MockData";
import Comment from "./Comment";

describe(`Comment Component`, () => {
  beforeAll(() => {
    AuthApiService.postUser(mockUser).then(user => {
      AuthApiService.postLogin({
        username: user.username,
        password: user.password
      });
    });

    ArticleApiService.postArticle(mockArticle).then(() => {
      CommentApiService.postComment(mockComment);
    });
  });

  it("renders Comment component without issues", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Comment comment={mockComment}></Comment>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
