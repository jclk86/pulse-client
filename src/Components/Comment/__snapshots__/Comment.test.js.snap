// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component renders Comment component without issues 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Comment)
      comment={
        Object {
          "article_id": 1,
          "content": "test comment content",
          "date_created": "2019-10-14T21:49:20.570Z",
          "id": 1,
          "user": Object {
            "id": 1,
            "username": "Pop123",
          },
        }
      }
    >
      <Comment
        comment={
          Object {
            "article_id": 1,
            "content": "test comment content",
            "date_created": "2019-10-14T21:49:20.570Z",
            "id": 1,
            "user": Object {
              "id": 1,
              "username": "Pop123",
            },
          }
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <li
          className="list_comment_item"
          key="1"
        >
          <div
            className="container_comment_content"
          >
            <span>
              Pop123
               says...
            </span>
             
            <p>
              October 14, 2019
            </p>
            <p>
              test comment content
            </p>
          </div>
        </li>
      </Comment>
    </withRouter(Comment)>
  </Router>
</BrowserRouter>
`;
